08_Interactive_Image_Visualization.ipynb

```diff
@@ -0,0 +1,1274 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, we will apply the same asinh stretch and zscale interval as was applied in `04_Intro_to_Butler` directly to the calexp object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Apply a asinh/zscale mapping to the data \n",
+    "transform = AsinhStretch() + ZScaleInterval()\n",
+    "scaledImage = transform(calexp.image.array)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define the (left, bottom, top, right) edges of the sensor image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bounds_img = (0, 0, calexp.getDimensions()[0], calexp.getDimensions()[1])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To use the matplotlib convention of placing the origin of an plot in the top left, we need to flip the data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scaledImage = np.flipud(scaledImage)\n",
+    "bounds_img = (0, calexp.getDimensions()[1], calexp.getDimensions()[0],0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define some default plot options for the Image\n",
+    "img_opts = dict(height=400, width=500, \n",
+    "                xaxis=\"bottom\", \n",
+    "                padding = 0.01, fontsize={'title': '8pt'},\n",
+    "                colorbar=True, toolbar='right', show_grid=True,\n",
+    "                tools=['hover']\n",
+    "               )     "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Make a function to generate a plot title from the dataId.\n",
+    "def dataIdToString(dataId: dict) -> str:\n",
+    "    title = \"DC2 image: \"\n",
+    "    for key, value in dataId.items():\n",
+    "        title += str(key) + \": \" + str(value) + \" \"\n",
+    "    return title.strip() "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create the Image element.\n",
+    "img = hv.Image(scaledImage, bounds=bounds_img,\n",
+    "               kdims=['x-axis', 'y-axis']).opts(\n",
+    "    cmap = bokeh.palettes.Viridis256, \n",
```

Comment:
Why Viridis and not gray scale to replicate what was done in the 03 notebooks?

Reply:
I played around with coour scales. Keith used Viridis in his original notebook. Further down in cell 38 I show using a grey scale with the source detections overlayed. I am happy to change the colour scale to whatever people prefer.

Reply:
I'd go with gray scale. I think it will look more "familiar" to astronomers.

Reply:
OK, I will change it.